src/mongo_odm/test/query_builder.cpp

```diff
@@ -150,6 +164,14 @@ TEST_CASE("Test nested member access.", "[mongo_odm::nvp_child]") {
         };
         test_lambda(MONGO_ODM_CHILD(BarParent, b, p, x));
     }
+
+    SECTION("Test nested members with optional") {
+        REQUIRE((MONGO_ODM_KEY(OptionalWithChildren::pt)->*MONGO_ODM_KEY(Point::x)).get_name() ==
+                "pt.x");
+        REQUIRE((MONGO_ODM_KEY(OptionalWithChildren::pt)->*MONGO_ODM_KEY(Point::x)).t == &Point::x);
+        REQUIRE(MONGO_ODM_CHILD(OptionalWithChildren, pt, x).get_name() == "pt.x");
+        REQUIRE(MONGO_ODM_CHILD(OptionalWithChildren, pt, x).t == &Point::x);
+    }
```

Comment:
Can you add a test for `std::vector<Point>` (eg) to show that the `iterable_value_t` is working? Also maybe a test for `optional<std::vector<Point>>` just for fun